base-ui/src/main/kotlin/com/tailoredapps/androidapptemplate/base/ui/ControllerViewModel.kt

```diff
@@ -29,7 +30,7 @@ abstract class ControllerViewModel<Action, State> : ViewModel() {
 
     fun dispatch(action: Action) = controller.dispatch(action)
     val currentState: State get() = controller.state.value
-    val state: Flow<State> get() = controller.state
+    val state: StateFlow<State> get() = controller.state
```

Comment:
Why expose the stateflow here?

Reply:
`StateFlow.collectAsState()` needs no initial value and uses the one specified during `StateFlow`s creation. The `controllers`s `StateFlow` is initialized with the `initialState` of the `Controller` and thus represents the actual state.

`Flow.collectAsState(initialValue)` requires an initial value. Specifying this initial value would lead to inconsistencies between the specified value and the actual state of the `Controller`

https://developer.android.com/reference/kotlin/androidx/compose/runtime/package-summary#(kotlinx.coroutines.flow.StateFlow).collectAsState(kotlin.coroutines.CoroutineContext)

Reply:
👌